src/platform/packages/shared/response-ops/alerts-filters-form/components/alerts_filter_by_rule_types.tsx

```diff
@@ -87,9 +92,21 @@ export const AlertsFilterByRuleTypes: AlertsFilterComponentType<string[]> = ({
   );
 };
 
-export const filterMetadata = {
+const isEmpty = (value?: string[]) => !Boolean(value?.length);
+
+export const filterMetadata: AlertsFilterMetadata<string[]> = {
```

Comment:
I initially typed these as
```ts
export const filterMetadata = {
  ...
} as const satisfies AlertsFilterMetadata<string[]>;
```
which is nice to have `id` typed as its literal value, but by typing the metadata object explicitly I can extract the type of the filter value more easily to avoid type casts (see [here](https://github.com/elastic/kibana/pull/216076/files#diff-ff37a7b35383b37b901c7f95c4755063dc81d7035f113527efccfcef32f7812eR27) for example).